e2etest/bgptests/bgp.go

```diff
@@ -98,9 +101,25 @@ var _ = ginkgo.Describe("BGP", func() {
 	})
 
 	ginkgo.BeforeEach(func() {
+
+		clientconfig := k8sclient.RestConfig()
+		var err error
```

Comment:
I had tod that because  otherwise it fails like https://github.com/metallb/metallb/actions/runs/9267228538/job/25493436433 (a wip commit that I reverse)

Reply:
There is some state that changes because I delete the speaker containers, so it must that logic go per test not per suite

Reply:
ah, I saw the reason. The FRR provider holds a reference to the speaker / frr-k8s pods.

This is a bit confusing though. FRRProvider is global, we initialize it two different times and we use it also in other files. I have the feeling this might bit us in the future (for example if we add a new context in a new file). 
How about giving the FRRProvider a Refresh() method that updates the speaker? So we re-load the proper speakers after any test that restarts the speakers?

Either that, or we fetch the speakers every time we call Execute. Shouldn't be terrible either as it fetches them from the local cache.

Reply:
I could go either, maybe the second seems simpler. Do you mean changing here https://github.com/metallb/metallb/blob/95274d94dffe0baed35330ebbbbe3f82282a19d1/e2etest/pkg/frr/provider/provider.go#L49 ? Can you expand?

Reply:
done